### PR TITLE
Add PropTypes and JSDocs to components

### LINKED ISSUE #139: Add propTypes and jsDocs for components
Use PropTypes(https://reactjs.org/docs/typechecking-with-proptypes.html) for components and JSDOCs(https://jsdoc.app/about-getting-started.html)for functions

Code to cover:

• src/app/component/PhoneLoginForm

• src/app/component/MapView

• src/app/component/Input

• src/app/component/ImageUpload

• src/app/component/FirebaseAuthUi

• src/app/component/MissionCard

• src/app/component/PrivateComponent

• src/app/component/ProtectedRoute

### PR DESCRIPTION
**Pull request by Eihcir0**
This pr adds PropTypes and JSDocs for components.  Closes #139 

Notes:
- I have never used JSDocs before, so I tried following the pattern I saw in the other PR -- let me know how I can improve on it
- I wasn't sure if PropTypes were needed on some of these smaller components like `PrivateComponent` and `ProtectedRoute` but I added them anyways for consistency and since they were part of the original request
- I fixed a bug I saw where we weren't handling if geoCoding did not return coordinates.  I added an early return and handling for this in `addressLookup.js` and `MissionDetailsContainer.js`
- ALSO NOTE: **We have 2 components named `MissionCard.js`** that are very similar. I was going to delete one or combine them but left them for now as I'm not sure if this was intentional

### CODE REVIEW

File: src/app/HOC/withLoading.js [hunk 1]
```diff
@@ -39,11 +39,11 @@ const withLoading = (BaseComponent) => {
   };
 
   EnhancedComponent.propTypes = {
+    children: PropTypes.any,
```
Comment (qdozaq) [inline src/app/HOC/withLoading.js:42]: You can use `PropTypes.element` here

File: src/app/component/ImageUpload/ImagePreview.js [hunk 2]
```diff
@@ -3,6 +3,11 @@ import PropTypes from "prop-types";
 import { ReactComponent as ClosePreviewIcon } from "../../../img/ClosePreview.svg";
 import { thumb, icon, img } from "./ImageUpload.style";
 
+/**
+ * Component for displaying image preview
+ *
+ * @component
+ */
```
Comment (qdozaq) [inline src/app/component/ImageUpload/ImagePreview.js:10]: We don't really need jsdocs for components just the propTypes

File: src/app/component/MapView/MapView.js [hunk 3]
```diff
@@ -14,11 +20,16 @@ function MapView(props) {
     <Map center={position} zoom={state.zoom} style={MapStyle} data-testid="map">
       <TileLayer url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" />
       <Marker position={position}>
-        <Popup>Start from here</Popup>
+        <Popup>Start from hereHOAGIES</Popup>
```
Comment (qdozaq) [inline src/app/component/MapView/MapView.js:23]: was this intentional? lol

File: src/app/component/Popup/Popup.jsx [hunk 4]
```diff
@@ -24,8 +29,9 @@ export default function Popup(props) {
 }
 
 Popup.propTypes = {
-  title: PropTypes.string,
-  handleClose: PropTypes.func,
   btnText: PropTypes.string,
+  children: PropTypes.any,
```
Comment (qdozaq) [inline src/app/component/Popup/Popup.jsx:33]: same here and a couple other places

File: src/app/component/MapView/MapView.js (hunk 3 above)
Comment (Eihcir0) [reply src/app/component/MapView/MapView.js:23]: hahahaha

**Comment by Eihcir0** [comment]
Thanks @qdozaq.  I will make the suggested changes and also create the new ticket for the duplicate components

**Pull request merged**